openedx/core/djangoapps/verified_track_content/partition_scheme.py

```diff
@@ -88,6 +90,8 @@ def get_group_for_user(cls, course_key, user, user_partition, **kwargs):  # pyli
                 mode_slug,
                 modes=CourseMode.modes_for_course(course_key, include_expired=True, only_selectable=False),
             )
+            if course_mode and CourseMode.is_credit_mode(course_mode):
```

Comment:
Yes, because "verified_mode_for_course" will return either Verified or Professional, depending on which mode that course has (preferring Verified).